chanbackup/recover.go

```diff
@@ -121,3 +129,122 @@ func UnpackAndRecoverMulti(packedMulti PackedMulti,
 
 	return Recover(chanBackups.StaticBackups, restorer, peerConnector)
 }
+
+// SignCloseTx produces a signed commit tx from a channel backup.
+func SignCloseTx(s Single, keyRing keychain.KeyRing, ecdher keychain.ECDHRing,
+	signer input.Signer) (*wire.MsgTx, error) {
+
+	if s.CloseTxInputs == nil {
+		return nil, errors.New("channel backup does not have data " +
+			"needed to sign force close tx")
+	}
+
+	// Each of the keys in our local channel config only have their
+	// locators populate, so we'll re-derive the raw key now.
+	var err error
+	s.LocalChanCfg.MultiSigKey, err = keyRing.DeriveKey(
+		s.LocalChanCfg.MultiSigKey.KeyLocator,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("unable to derive multisig key: %w", err)
+	}
+
+	signDesc, err := createSignDesc(s)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create signDesc: %w", err)
+	}
+
+	inputs := lnwallet.SignedCommitTxInputs{
+		CommitTx:  s.CloseTxInputs.CommitTx,
+		CommitSig: s.CloseTxInputs.CommitSig,
+		OurKey:    s.LocalChanCfg.MultiSigKey,
+		TheirKey:  s.RemoteChanCfg.MultiSigKey,
+		SignDesc:  signDesc,
+	}
+
+	if s.Version == SimpleTaprootVersion {
```

Comment:
Same here re adding a helper method `s.version.IsTaproot()` to also catch future overlay channel types. Here and in `createSignDescs`.

Reply:
Done.